### PR TITLE
Fallback to cookie generation if CID API failed in anyway.

### PR DESCRIPTION
**Pull request by lannka**

### CODE REVIEW

File: src/service/cid-api.js [hunk 1]
```diff
@@ -87,7 +89,7 @@ export class GoogleCidApi {
       return token !== TokenStatus.RETRIEVING;
     }).then(() => {
       if (token === TokenStatus.OPT_OUT) {
-        return null;
+        return OPT_OUT;
```
Comment (cramforce) [inline src/service/cid-api.js]: Use the TokenStatus instead of new global?

File: src/service/cid-impl.js [hunk 2]
```diff
@@ -181,7 +181,12 @@ export class Cid {
           ViewerCidApi.scopeOptedInForCidApi(this.ampdoc.win, scope);
       if (apiClient) {
         return this.cidApi_.getScopedCid(
-            apiClient, scope, getCidStruct.cookieName);
+            apiClient, scope, getCidStruct.cookieName).then(scopedCid => {
+              if (scopedCid) {
+                return (scopedCid == OPT_OUT) ? null : scopedCid;
+              }
+              return getOrCreateCookie(this, getCidStruct, persistenceConsent);
```
Comment (cramforce) [inline src/service/cid-impl.js]: You could now delete the `getCookie` calls (e.g. line 101) for fallback in cid-api, right?

**Comment by lannka** [comment]
@cramforce refactored code a bit. moved out all cookie logic back to cid-impl

File: test/functional/test-cid-api.js [hunk 3]
```diff
@@ -248,7 +248,7 @@ describes.realWin('test-cid-api', {}, env => {
   it('should return null if AMP_TOKEN=$OPT_OUT ', () => {
     persistCookie('AMP_TOKEN', '$OPT_OUT');
     return api.getScopedCid('googleanalytics', 'scope-a').then(cid => {
-      expect(cid).to.be.null;
+      expect(cid).to.equal('$OPT_OUT');
```
Comment (avimehta) [inline test/functional/test-cid-api.js]: With this change, can you add a test to make sure `getExternalCid_` still returns null for client id?

Comment (lannka) [inline test/functional/test-cid-api.js]: a unit test exists: https://github.com/lannka/amphtml/blob/b75e04837cb44918a509285ef131e925252cce40/test/functional/test-cid.js#L817

also tested manually by changing the browser cookie to $OPT_OUT

Comment (avimehta) [inline test/functional/test-cid-api.js]: Thanks!

Comment (lannka) [inline test/functional/test-cid-api.js]: @avimehta thank you for the late night review!
